### PR TITLE
fix: Remove a few panics

### PR DESCRIPTION
**Pull request by huitseeker**
Those panics should not crash the node. 
Folding them in the error case of their immediately enclosing `Result`.

### CODE REVIEW

**Review by gdanezis — APPROVED** [review]
Many thanks for tidying these!

**Review by joyqvq — APPROVED** [review]
💯

File: crates/sui-types/src/crypto.rs [hunk 1]
```diff
@@ -605,7 +605,8 @@ where
 {
     fn from_signable_bytes(bytes: &[u8]) -> Result<Self, Error> {
         // Remove name tag before deserialization using BCS
-        let name = serde_name::trace_name::<Self>().expect("Self must be a struct or an enum");
```
Comment (bmwill) [inline crates/sui-types/src/crypto.rs:608]: this should 100% be a panic

File: crates/sui-sdk/src/crypto.rs [hunk 2]
```diff
@@ -135,6 +135,9 @@ impl SuiKeystoreSigner {
 
 impl signature::Signer<Signature> for SuiKeystoreSigner {
     fn try_sign(&self, msg: &[u8]) -> Result<Signature, signature::Error> {
-        self.keystore.read().unwrap().sign(&self.address, msg)
```
Comment (bmwill) [inline crates/sui-sdk/src/crypto.rs:138]: This should be a panic, as the only time this unwrap would trigger would be on a poisoned locked which we have no way of recovering from

**Review by bmwill — CHANGES_REQUESTED** [review]
We can't just convert panic to errors in all cases, even though functions return Result.

File: crates/sui-types/src/crypto.rs (hunk 1 above)
Comment (huitseeker) [reply crates/sui-types/src/crypto.rs:608]: This function is part of a blanket implementation pegged on a marker trait.
If anyone slaps this marker trait on something that is not `serde_name`-able (more specifically `serde_name`-traceable), I have a problem with the delay between that mistake slipping in and the much later time at which we will discover asking for signature of any such object crashes the node. This is a textbook case of remote-crashing a family of nodes.

File: crates/sui-sdk/src/crypto.rs (hunk 2 above)
Comment (huitseeker) [reply crates/sui-sdk/src/crypto.rs:138]: Fair enough, my panic aversion should not extend to client binaries.

File: crates/sui-types/src/crypto.rs (hunk 1 above)
Comment (huitseeker) [reply crates/sui-types/src/crypto.rs:608]: Ah, never mind, there's no need to remove the panic there, I should use a sealed trait in place of this `BCSSignable` 🤦 
To be continued

**Review by bmwill — APPROVED** [review]

**Pull request merged**